lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -72,6 +74,45 @@ class FullPageScreenshot extends Gatherer {
     };
   }
 
+  /**
+   * Gatherers can collect details about DOM nodes, including their position on the page.
+   * Layout shifts occuring after a gatherer runs can cause these positions to be incorrect,
+   * resulting in a poor experience for element screenshots.
+   * `getNodeDetails` maintains a collection of DOM objects in the page, which we can iterate
+   * to re-collect the bounding client rectangle. We also update the devtools node path.
+   * The old devtools node path is used as a lookup key. We walk the entire artifacts object
+   * to update all objects that reference an old devtools node path.
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts.FullPageScreenshot['nodes']>}
+   */
+  async _resolveNodes(passContext) {
+    if (!process.env.RESOLVE_NODES) return {};
+
+    function resolveNodes() {
+      /** @type {LH.Artifacts.FullPageScreenshot['nodes']} */
+      const nodes = {};
+      if (!window.__lighthouseNodesDontDeleteOrYoureFired) return nodes;
+
+      for (const [id, node] of window.__lighthouseNodesDontDeleteOrYoureFired.entries()) {
```

Comment:
Not sure if there is a useful unit test to make for `_resolveNodes`. It'd just be mocking calls to RuntimeEvaluate and making sure the results of two calls are spread together... not especially useful?

I added expectations to the screenshot smoke test. Is that enough?

Reply:
@patrickhulce @adamraine  wdyt